### PR TITLE
Feature/carpark admin panel

### LINKED ISSUE #44: Carpark Backend Form [Booking Sys Func]

### PR DESCRIPTION
**Pull request by zollf**
Kinda went overboard lol

I added a custom tag to django templates that will build the tables and forms for us. 

closes #44

### CODE REVIEW

File: app/admin/templatetags/admin.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from django import template
+
+register = template.Library()
+
+@register.inclusion_tag('table.html')
+def create_table(title: str, data: dict, headers: str):
+    """
+    Creates table given object and headers
+    Usage: `{% create_form 'carparks' carpark 'name|description' %}`
```
Comment (APM246) [inline app/admin/templatetags/admin.py]: Should this be Usage: `{% create_table ...   instead?

File: app/resources/templates/user.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+{% block admin_head %}
+ {% load static %}
+  <link rel="stylesheet" type="text/css" href="{% static 'styles/admin_buttons.css' %}">
+  <script src="{% static '/scripts/delete.js' %}"></script>
+{% endblock %}
+
+{% block admin_content %}
+  {% create_form 'users' user 'username:text|password:password' %}
+{% endblock %}
```
Comment (APM246) [inline app/resources/templates/user.html:12]: Should the users/view/id page have a delete button similar to carparks/view/id?

**Review by APM246 — CHANGES_REQUESTED** [review]

**Comment by APM246** [comment]
Perhaps the parking folder in app should be within a folder "api" for clarity

**Comment by APM246** [comment]
Just realised modifying a carpark ends up adding a new entry instead, is this a bug or because the page is still a work in progress?

**Comment by APM246** [comment]
Does the code ensure API calls can only be made by authenticated users?  
https://www.django-rest-framework.org/api-guide/views/#api-policy-decorators and 
https://www.django-rest-framework.org/api-guide/settings/ discusses how to restrict API calls.

**Comment by zollf** [comment]
> 
> 
> Does the code ensure API calls can only be made by authenticated users?
> https://www.django-rest-framework.org/api-guide/views/#api-policy-decorators and
> https://www.django-rest-framework.org/api-guide/settings/ discusses how to restrict API calls.

so we can actually just do `@login_required(login_url="/login")` which will send unauthorized people to login https://gyazo.com/979fb41e6083803584036be5e938c3b4. I'll probably just do that for know, as he hasn't spent much on security .

**Comment by zollf** [comment]
> 
> 
> Just realised modifying a carpark ends up adding a new entry instead, is this a bug or because the page is still a work in progress?

🙇

**Comment by zollf** [comment]
> 
> 
> Perhaps the parking folder in app should be within a folder "api" for clarity

we should probably leave this as everyone will then have to do the migrations, which is a hassle. We'll add the newer stuff to the folder api, and bays will just sit in parking.

File: app/resources/templates/user.html (hunk 2 above)
Comment (zollf) [reply app/resources/templates/user.html:12]: yea itll have the same buttons, there are other buttons that we will add later, or can just find a way to optionally add them in.

**Review by APM246 — COMMENTED** [review]
Modifying a car park still ends up adding a new car park rather than modifying the current details.

**Review by APM246 — APPROVED** [review]
Looks good

**Pull request merged**